### ISSUE TITLE
WCAG Section 3.3.2 - User input incomplete in field.

### ISSUE THREAD

**Issue by jonrussellsoftwaretester** [jonrussellsoftwaretester/508-Tools-Team#4]
WCAG Section 3.3.2 - User input incomplete in field.

[Error Suggestion: If an input error is automatically detected and suggestions for correction are known, then the suggestions are provided to the user, unless it would jeopardize the security or purpose of the content. The intent of this Success Criterion is to ensure that users receive appropriate suggestions for correction of an input error if it is possible. The definition of "input error" says that it is "information provided by the user that is not accepted" by the system. Some examples of information that is not accepted include information that is required but omitted by the user and information that is provided by the user but that falls outside the required data format or allowed values.]

Labels or instructions are provided when content requires user input.

There are several techniques you should employ when creating a form to make it WCAG accessible:

Provide descriptive labels. This has been discussed earlier in the WCAG guide where you describe the purpose of each form element using a <label> tag.

Providing text instructions at the beginning of a form or set of fields that describes the necessary input. For example, at the beginning of the form you should specify to the user how dates should be entered ("Dates should be entered in mm/dd/yyyy format.")

You can also place a specific format instruction in a particular field's label. For example:

<label for="date">Date (mm/dd/yyyy)</label>

<input type="text" name="date" id="date" />

Place checkboxes and radio buttons before their labels, not after. Another obvious point is to place the labels for text input before the text input and not after.

Indicating a required field with text in the label. Simply adding the phrase "(required)" in the label for your field will be adequate.
Indicating an uncompleted field that was required after submission. For example, if someone's name was required and they didn't complete it, you can modify the label to say "(not completed)" in addition to its name. This isn't necessary but it helps.

Finally, like mentioned before in this guide, you can use the title attribute to name the purpose of a field when using a label isn't possible.

<input type="text" name="date" id="date" title="Date (required)" />
